api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1248,9 +1248,104 @@ const (
+	// +kubebuilder:validation:Enum=Enabled;Disabled
+	// +optional
+	Enabled string `json:"enabled,omitempty"`
+
+	// delayAfterAddSeconds sets how long after scale up the scale down evaluation resumes in seconds
+	// It must be between 0 and 86400.
+	// When set to 0, this means scale down evaluation will resume immediately after scale up, without any delay.
+	// When omitted, this means no opinion, and the platform is left to choose a reasonable default.
+	//
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=86400
+	// +optional
+	DelayAfterAddSeconds *int32 `json:"delayAfterAddSeconds,omitempty"`
+
+	// delayAfterDeleteSeconds sets how long after node deletion, scale down evaluation resumes, defaults to scan-interval
+	// It must be between 0 and 86400.
+	// When set to 0, this means scale down evaluation will resume immediately after node deletion, without any delay.
+	// When omitted, this means no opinion, and the platform is left to choose a reasonable default.
+	// The default value is 60.
+	//
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=86400
+	// +kubebuilder:default=60
+	// +optional
+	DelayAfterDeleteSeconds *int32 `json:"delayAfterDeleteSeconds,omitempty"`
+
+	// delayAfterFailureSeconds sets how long after a scale down failure, scale down evaluation resumes
+	// It must be between 0 and 86400.
+	// When set to 0, this means scale down evaluation will resume immediately after a scale down failure, without any delay.
+	// When omitted, this means no opinion, and the platform is left to choose a reasonable default.
+	// The default value is 60.
+	//
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=86400
+	// +kubebuilder:default=60
+	// +optional
+	DelayAfterFailureSeconds *int32 `json:"delayAfterFailureSeconds,omitempty"`
+
+	// unneededDurationSeconds establishes how long a node should be unneeded before it is eligible for scale down in seconds
+	// It must be between 0 and 86400.
+	// When set to 0, this means nodes will be considered for scale down immediately when they become unneeded, without any waiting period.
+	// When omitted, this means no opinion, and the platform is left to choose a reasonable default.
+	// The default value is 600.
+	//
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=86400
+	// +kubebuilder:default=600
+	// +optional
+	UnneededDurationSeconds *int32 `json:"unneededDurationSeconds,omitempty"`
+
+	// utilizationThreshold determines the node utilization level, defined as sum of requested resources divided by capacity, below which a node can be considered for scale down
+	// The value represents a percentage from 0 to 100.
+	// When set to 0, this means nodes will be considered for scale down even if they are completely idle (0% utilization).
+	// When set to 100, this means nodes will only be considered for scale down when they are fully utilized (100% utilization).
```

Comment:
I don't think these descriptions are correct.

IIRC, the utilization threshold is an upper bound for consideration, you've described it as a lower bound.

So a 0% means the node would have to be completely empty to be considered for scale down.

And 100% means that they will be considered for scale down always, even if they are 100% utilised.

A 50% threshold means they will be considered if their utilisation is up to, and including 50%, but a node using 51% would not be considered.

Right?